addons/web_editor/static/src/js/editor/odoo-editor/src/commands/enter.js

```diff
@@ -91,8 +91,9 @@ HTMLElement.prototype.oEnter = function (offset, firstSplit = true) {
  */
 HTMLHeadingElement.prototype.oEnter = function () {
     const newEl = HTMLElement.prototype.oEnter.call(this, ...arguments);
-    if (!newEl.textContent) {
+    if ([...newEl.textContent].every(char => char === '\u200B')) { // empty or all invisible
```

Comment:
I think we do have an util that should do the trick here.

Reply:
Which one? Can't think of one that does this.

Reply:
Wouldn't `isVisibleStr` work in your case ?

Reply:
Ah I guess it would! `if (!isVisibleStr(newEl.textContent)) {`